src/ApiHandler.ts

```diff
@@ -139,6 +125,69 @@ export default class ApiHandler {
 			}
 		}
 
+		const byteFee = api.consts.transactionPayment.transactionByteFee;
+		const extrinsicBaseWeight = api.consts.system.extrinsicBaseWeight;
+		const coefficients = api.consts.transactionPayment.weightToFee[0];
+		const coeffFrac = coefficients.coeff_frac;
+		const coeffInt = coefficients.coeff_integer;
+		
+		// TODO: generalize for terms that may have more degrees & may be negative
+		const degree = coefficients.degree;
+		const negative = coefficients.negative;
+
+		for (let idx = 0; idx < block.extrinsics.length; ++idx) {
+			const extrinsic = block.extrinsics[idx];
+			if (!extrinsic.isSigned) {
+				continue;
+			}
+
+			try {
+				if (api.runtimeVersion.specName.toString() === 'kusama') {
```

Comment:
We're making the assumption that sidecar is only running against kusama and polkadot, right?

Reply:
No, Kusama doesn't have the polynomial WeightToFee, so we still have to (incorrectly) use `queryInfo`. But since this code is mostly generic (assuming 1 coefficient of degree 1) then it will work on any chain built on Substrate master, including Polkadot. Once Kusama gets a runtime upgrade to latest Substrate, then we can pull this out.

Reply:
> (assuming 1 coefficient of degree 1)

It is not difficult to remove this assumption, though. Just a loop over all coefficients. But that can be a followup PR in order to get this working asap.

Reply:
Correct, I was talking more about removing the `kusama` bit. Once Kusama gets an upgrade and we loop over all coefficients here, it will be a clean solution.